### PR TITLE
Add possibility to send telemetry events by data collectors

### PR DESCRIPTION
**Pull request by jakubch1**
## Description

Please add a meaningful description for this change.
Ensure the PR has required unit tests.

## Related issue

Kindly link any related issues. E.g. Fixes #xyz.

- [ ] I have ensured that there is a previously discussed and approved issue.

### CODE REVIEW

**Review by cvpoienaru — APPROVED** [review]
LGTM

File: src/Microsoft.TestPlatform.ObjectModel/ITelemetryInitializer.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+/// <summary>
+/// Interface for extensions that choose to send telemetry events
+/// </summary>
+public interface ITelemetryInitializer
+{
+    /// <summary>
+    /// Initializes telemetry reporter
+    /// </summary>
+    void Initialize(ITelemetryReporter telemetryReporter);
```
Comment (MarcoRossignoli) [inline src/Microsoft.TestPlatform.ObjectModel/ITelemetryInitializer.cs:14]: I wonder if we should have an option object for future extensibility but maybe we can add to the `ITelemetryReporter` self if needed

**Review by MarcoRossignoli — APPROVED** [review]
some comment but I don't know if we can do better

File: src/Microsoft.TestPlatform.ObjectModel/ITelemetryInitializer.cs (hunk 1 above)
Comment (jakubch1) [reply src/Microsoft.TestPlatform.ObjectModel/ITelemetryInitializer.cs:14]: I think extending API for extensions is not issue anymore.

File: src/Microsoft.TestPlatform.ObjectModel/TelemetryEvent.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+namespace Microsoft.VisualStudio.TestPlatform.ObjectModel;
+
+public sealed class TelemetryEvent
+{
+    /// <summary>
+    /// Initialize an TelemetryEvent
+    /// </summary>
+    /// <param name="name">Telemetry event name</param>
+    /// <param name="properties">Telemetry event properties</param>
+    public TelemetryEvent(string name, IDictionary<string, object> properties)
```
[the remark this answers is not in the document]
Comment (jakubch1) [reply src/Microsoft.TestPlatform.ObjectModel/TelemetryEvent.cs:16]: this is API used by VS telemetry

**Pull request merged**